derive-error-chain-tests/src/main.rs

```diff
@@ -511,8 +511,11 @@ mod generics_test {
 		#[error_chain(link = "inner2::Error")]
 		Link(inner2::ErrorKind),
 
-		// FIXME: conflicting implementations of trait `std::convert::From<&str>` for type `generics_test::Error<&str, _>
-		// FIXME: conflicting implementations of trait `std::convert::From<std::string::String>` for type `generics_test::Error<std::string::String, _>
+		// FIXME: conflicting implementations of trait `std::convert::From<generics_test::inner1::Error<_>>` for type `generics_test::Error<generics_test::inner1::Error<_>, _>`
```

Comment:
Indeed, the conflict is between the From impl for LinkGeneric and ForeignGeneric.

```rust
impl <T: error::Error + Send + 'static, U> From<inner1::Error<U>> for Error<T, U>
    where U: Send + fmt::Debug + fmt::Display + 'static {

    fn from(err: inner1::Error<U>) -> Self {
        Error(ErrorKind::LinkGeneric(err.0), err.1)
    }
}
```

and

```rust
impl <T: error::Error + Send + 'static, U> From<T> for Error<T, U>
    where U: Send + fmt::Debug + fmt::Display + 'static {

    fn from(err: T) -> Self {
        Self::from_kind(ErrorKind::ForeignGeneric(err))
    }
}
```